src/libraries/System.Private.CoreLib/src/System/IO/Strategies/WindowsFileStreamStrategy.cs

```diff
@@ -165,33 +182,34 @@ public sealed override long Seek(long offset, SeekOrigin origin)
             if (_fileHandle.IsClosed) ThrowHelper.ThrowObjectDisposedException_FileClosed();
             if (!CanSeek) ThrowHelper.ThrowNotSupportedException_UnseekableStream();
 
-            // Verify that internal position is in sync with the handle
-            VerifyOSHandlePosition();
-
             long oldPos = _filePosition;
-            long pos = SeekCore(_fileHandle, offset, origin);
+            long pos = origin switch
+            {
+                SeekOrigin.Begin => offset,
+                SeekOrigin.End => FileStreamHelpers.GetFileLength(_fileHandle, _path) + offset,
```

Comment:
SeekForward is now x20 times faster, SeekBackward just 24%. I am going to merge the PR, but could you please verify if it would be possible to use `Length` here and send another PR if it works fine? If we can we should take advantage of the caching you have added.

```suggestion
                SeekOrigin.End => Length + offset,
```

Reply:
Good point.

Reply:
Created https://github.com/dotnet/runtime/pull/50218 for this.